x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/generate_api_key_modal/modal.tsx

```diff
@@ -0,0 +1,173 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiCopy,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiFieldText,
+  EuiFormRow,
+  EuiText,
+  EuiSpacer,
+  EuiLink,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { GenerateApiKeyModalLogic } from './generate_api_key_modal.logic';
+
+interface GenerateApiKeyModalProps {
+  indexName: string;
+  onClose(): void;
+}
+
+export const GenerateApiKeyModal: React.FC<GenerateApiKeyModalProps> = ({ indexName, onClose }) => {
+  const { keyName, apiKey, isLoading, isSuccess } = useValues(GenerateApiKeyModalLogic);
+  const { setKeyName, makeRequest } = useActions(GenerateApiKeyModalLogic);
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {i18n.translate('xpack.enterpriseSearch.content.overview.generateApiKeyModal.title', {
+            defaultMessage: 'Generate API Key',
+          })}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <>
+          <EuiText size="m">
+            <p>
+              {i18n.translate('xpack.enterpriseSearch.content.overview.generateApiKeyModal.info', {
+                defaultMessage:
+                  "Before you can start posting documents to your Elasticsearch index you'll need to create at least one API key.",
+              })}
+              &nbsp;
+              <EuiLink href={/* TODO link to docs */ '#'} external>
+                {i18n.translate(
+                  'xpack.enterpriseSearch.content.overview.generateApiKeyModal.learnMore',
+                  { defaultMessage: 'Learn more about API keys' }
+                )}
+              </EuiLink>
+            </p>
+          </EuiText>
+          <EuiSpacer />
+          <EuiPanel hasShadow={false} color="primary">
+            <EuiFlexGroup direction="column">
+              <EuiFlexItem>
+                <EuiFlexGroup direction="row" alignItems="center">
+                  <EuiFlexItem>
+                    {!isSuccess ? (
+                      <EuiFormRow label="Name your API key">
+                        <EuiFieldText
+                          placeholder="Type a name for your API key"
+                          onChange={(event) => setKeyName(event.currentTarget.value)}
+                          isLoading={isLoading}
+                        />
+                      </EuiFormRow>
+                    ) : (
+                      <EuiFormRow label={`"${keyName.trim()}"`}>
+                        <EuiFieldText value={apiKey} />
+                      </EuiFormRow>
+                    )}
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <EuiCopy
```

Comment:
You can use EuiCode for this, or even better: our own Enterprise Search ApiKey component.